### DISCUSSION TITLE
Is there anyway to check if a string contains emojis?

### DISCUSSION THREAD

**Discussion by JJPell** [colinhacks/zod#2848] [Q&A] [answered]
I'm currently using:

```javascript
z.string()
 .regex(/^((?!\p{Extended_Pictographic}).)*$/u, 'Emojis are not allowed');
```

Just wondered if there was another way of doing this?

**Comment by lukecarr** [comment, marked as answer]
Depending on your use case, it might be worth using `Emoji_Presentation` (rather than `Extended_Pictographic`).

Extended pictographic will match all emojis (that appear as "textured" emojis) as well as other "pictographic" characters like ⌨︎.

Emoji presentation, however, will only match your typical "emojis".

`Extended_Pictographic` could be what you need, but it's a useful distinction to know about. This [SO answer](https://stackoverflow.com/a/72727900) explains further.

**Comment by JJPell** [comment]
Thanks, that's super useful 👍

**Comment by rvijay17** [comment]
Is there a way to use the z.string().emoji() type by creating an intersection between string and NOT emoji? I just don't wanna use regex when there's a type for emojis already.
